File: src/vsb/graphinterfaces/Vertex.java
package vsb.graphinterfaces;

import java.util.Set;

public interface Vertex {
	
	long getId();
	void addEdge(Edge edge);
	Metadata getMetadata();
	Set<Edge> getEdges();
	Set<Edge> getEdgesBy(EdgeFilter filter);
	Set<Vertex> getAdjacentVertices();
	Set<Vertex> getAdjacentVertices(VertexFilter filter);
	
}

File: src/vsb/graphinterfaces/Vertex.scala
package vsb.graphinterfaces

trait Vertex {
	
	def getId: Long
	def addEdge(edge: Edge)
	def getMetadata: Metadata
	def getEdges: scala.collection.Set[Edge]
	def getEdgesBy(filter: EdgeFilter): scala.collection.Set[Edge]
	def getAdjacentVertices: scala.collection.Set[Vertex]
	def getAdjacentVertices(filter: VertexFilter): scala.collection.Set[Vertex]

}
